storage/rocksdb/ha_rocksdb.cc

```diff
@@ -10775,6 +11271,8 @@ int ha_rocksdb::check_and_lock_sk(
     lock for this key.
   */
   if (!(key_info->flags & HA_NOSAME)) {
+    if (rocksdb_use_range_locking)
+      return check_and_lock_non_unique_sk(key_id, row_info);
```

Comment:
Maybe add a comment about why we don't use m_use_range_locking instead? I assume it's because we still want RC transactions to be blocked by gap locks held by RR transactions.

Reply:
Testcases confirm that this is indeed the reason. Will need to check whether the plain `rocksdb_use_range_locking` check is not too broad regarding temporary, DD etc. tables.